src/main/java/uk/gov/hmcts/reform/profilesync/config/SecurityConfiguration.java

```diff
@@ -2,74 +2,113 @@
 
-                                 RequestAuthorizer<Service> serviceRequestAuthorizer,
+    @Value("${spring.security.oauth2.client.provider.oidc.issuer-uri}")
+    private String issuerUri;
 
-                                 AuthenticationManager authenticationManager) {
+    @Value("${oidc.issuer}")
+    private String issuerOverride;
 
-        authCheckerServiceAndUserFilter = new AuthCheckerServiceAndUserFilter(serviceRequestAuthorizer, userRequestAuthorizer);
+    private  ServiceAuthFilter serviceAuthFilter;
+    List<String> anonymousPaths;
 
-        authCheckerServiceAndUserFilter.setAuthenticationManager(authenticationManager);
-
-    }
+    private JwtAuthenticationConverter jwtAuthenticationConverter;
 
     public List<String> getAnonymousPaths() {
         return anonymousPaths;
     }
 
+    public void setAnonymousPaths(List<String> anonymousPaths) {
+        this.anonymousPaths = anonymousPaths;
+    }
+
     @Override
     public void configure(WebSecurity web) {
         web.ignoring()
-                .antMatchers(anonymousPaths.toArray(new String[0]));
+                    .antMatchers(anonymousPaths.toArray(new String[0]));
+    }
+
+    @Inject
+    public SecurityConfiguration(final JwtGrantedAuthoritiesConverter jwtGrantedAuthoritiesConverter,
+                                                       final ServiceAuthFilter serviceAuthFilter
+    ) {
+
+        this.serviceAuthFilter = serviceAuthFilter;
+        jwtAuthenticationConverter = new JwtAuthenticationConverter();
+        jwtAuthenticationConverter.setJwtGrantedAuthoritiesConverter(jwtGrantedAuthoritiesConverter);
+
     }
 
     @Override
-    protected void configure(HttpSecurity http) throws Exception {
-
-        http.authorizeRequests()
-                .antMatchers("/actuator/**")
-                .permitAll()
-                .and()
-                .sessionManagement()
-                .sessionCreationPolicy(STATELESS)
-                .and()
-                .csrf()
-                .disable()
-                .formLogin()
-                .disable()
-                .logout()
-                .disable()
-                .authorizeRequests()
-                .anyRequest()
-                .authenticated()
-                .and()
-                .addFilter(authCheckerServiceAndUserFilter);
+    protected void configure(final HttpSecurity http) throws Exception {
+        http
+           .addFilterBefore(serviceAuthFilter, BearerTokenAuthenticationFilter.class)
+           .sessionManagement().sessionCreationPolicy(STATELESS).and()
+           .csrf().disable()
+           .formLogin().disable()
+           .logout().disable()
+           .authorizeRequests()
+           .antMatchers("/error").permitAll()
+           .antMatchers(HttpMethod.POST, "/refdata/external/v1/organisations").permitAll()
+           .antMatchers(HttpMethod.POST, "/refdata/internal/v1/organisations").permitAll()
+           .anyRequest()
```

Comment:
are these urls correct here? it is doing a match on v1/organisations

Reply:
it is not required here ..removed the changes